clients/pkg/promtail/svc/service_windows.go

```diff
@@ -0,0 +1,44 @@
+//+build windows
+
+package svc
+
+import (
+	"time"
+
+	wsvc "golang.org/x/sys/windows/svc"
+)
+
+type svcWindows struct{}
+
+func init() {
+	interactive, err := wsvc.IsAnInteractiveSession()
+	if err != nil {
+		panic(err)
+	}
+	if interactive {
+		return
+	}
+	go func() {
+		_ = wsvc.Run("", svcWindows{})
+	}()
+}
+
+func (svcWindows) Execute(args []string, r <-chan wsvc.ChangeRequest, s chan<- wsvc.Status) (ssec bool, errno uint32) {
+	const accCommands = wsvc.AcceptStop | wsvc.AcceptShutdown
+	s <- wsvc.Status{State: wsvc.StartPending}
+	s <- wsvc.Status{State: wsvc.Running, Accepts: accCommands}
+	for {
+		c := <-r
+		switch c.Cmd {
```

Comment:
Shouldn't the logic be more complex ? Like stopping promtail ? Actually I have no idea of that works, I'd love to learn more about.